Sources/SpeziViews/Views/Layout/HorizontalGeometryReader.swift

```diff
@@ -38,20 +38,8 @@ public struct HorizontalGeometryReader<Content: View>: View {
                         .preference(key: WidthPreferenceKey.self, value: geometry.size.width)
                 }
             )
-            .onPreferenceChange(WidthPreferenceKey.self) { width in
-                // The `onPreferenceChange` view modfier now takes a `@Sendable` closure, therefore we cannot capture `@MainActor` isolated properties
-                // on the `View` directly anymore: https://developer.apple.com/documentation/swiftui/view/onpreferencechange(_:perform:)?changes=latest_minor
-                // However, as the `@Sendable` closure is still run on the MainActor (at least in my testing on 18.2 RC SDKs), we can use `MainActor.assumeIsolated`
-                // to avoid scheduling a `MainActor` `Task`, which could delay execution and cause unexpected UI behavior.
-                if Thread.isMainThread {
-                    MainActor.assumeIsolated {
-                        self.width = width
-                    }
-                } else {
-                    Task { @MainActor in
-                        self.width = width
-                    }
-                }
+            .onPreferenceChange(WidthPreferenceKey.self) { [$width] width in
```

Comment:
I'm somewhat amazed that this is actually possible, capturing the `Binding` of the `MainActor` isolated `width` property (`Sendable` conformance, so fine) and then **mutating** the **wrapped value** (which should be `MainActor` isolated) on an **arbitrary actor** (`onPreferenceChange()` is for now called on the `MainActor` but that may change in the future).

These conformances in SwiftUI make that capture actually possible:
```swift
extension Binding : @unchecked Sendable where Value : Sendable { }

extension CGFloat : Sendable { }
```

Are we really sure that this isn't just a Swift compiler isolation detection miss or something? Seems a bit strange to me. 
~~My guess would be that this crashes at runtime as soon as `onPreferenceChange()` invocation moves off the `MainActor`?)~~

EDIT: Just saw that you actually tested this in your little code snipped:
```swift
.onPreferenceChange(WidthPreferenceKey.self) { [$width] width in
    dispatchPrecondition(condition: .onQueue(.main))
    Task.detached {
        dispatchPrecondition(condition: .notOnQueue(.main))
        $width.wrappedValue = width // works fine
    }
}
```
Still, I'm not really convinced that we're doing the right thing here?
@Supereg What's your opinion?

Reply:
@philippzagar tbh i also was kinda suspicious at first (my initial local workaround was a lot uglier, until i eventually realised that i could simply capture the binding direcly), but i came across [this SO answer](https://stackoverflow.com/a/79273163/2513803) which claims, albeit without any citations, that this approach is apple-sanctioned

my guess is that `Binding.wrappedValue.set` simply does the "jump onto main actor if necessary" thing internally?

Reply:
Since I merged #48, there (thankfully) have been more discussions about the `@Sendable` preference closure change! Thanks for finding that SO post.

The author of the comment (claiming Apple sanctioned this code piece) seems somewhat credible to me: https://stackoverflow.com/users/3791229/smr

Still, I wouldn't just blindly follow this code piece, just because "it works" (for now).
I don't even want to guess what `Binding.wrappedValue.set` then does internally.. and if it will change rather sooner than later.

Reply:
Tbh, I think we don't even need to deviate from the current implementation that checks if we're on the mainactor and then assumes that we are. From your description (and reply here https://github.com/StanfordSpezi/SpeziViews/pull/51#discussion_r1899724518), that implementation didn't even break anything. Of course, sooner or later, we need a better, more permanent solution here (especially when SwiftUI moves the invocation off the main actor)